hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/DeserializedGenericRecordBuilder.java

```diff
@@ -32,27 +33,34 @@
  */
 public class DeserializedGenericRecordBuilder extends AbstractGenericRecordBuilder {
 
-    private final TreeMap<String, Object> objects = new TreeMap<>();
+    private final TreeMap<String, Object> objects;
     private final SchemaWriter schemaWriter;
+    private boolean built;
 
     public DeserializedGenericRecordBuilder(String typeName) {
+        this.objects = new TreeMap<>();
         schemaWriter = new SchemaWriter(typeName);
     }
 
     /**
      * @return newly created GenericRecord
      * @throws HazelcastSerializationException if a field is not written when building with builder from
-     *                                         {@link GenericRecordBuilder#compact(String)} (ClassDefinition)} and
+     *                                         {@link GenericRecordBuilder#portable(ClassDefinition)} and
      *                                         {@link GenericRecord#newBuilder()}
      */
     @Nonnull
     @Override
     public GenericRecord build() {
-        return new DeserializedGenericRecord(schemaWriter.build(), objects);
+        GenericRecord record = new DeserializedGenericRecord(schemaWriter.build(), objects);
+        this.built = true;
+        return record;
     }
 
 
     protected GenericRecordBuilder write(@Nonnull String fieldName, Object value, FieldKind fieldKind) {
+        if (this.built) {
+            throw new HazelcastSerializationException("Cannot modify the generic record after building");
```

Comment:
I think the error message is a bit misleading: We are not modifying the generic record, rather, we are trying to modify the generic record builder, after building the generic record

Reply:
Ok let me make it: 

Cannot modify the GenericRecordBuilder after building